ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/mainnet/MainnetProtocolSpecs.java

```diff
@@ -59,7 +60,8 @@
 
   private MainnetProtocolSpecs() {}
 
-  public static ProtocolSpecBuilder<Void> frontierDefinition() {
+  public static ProtocolSpecBuilder<Void> frontierDefinition(final OptionalInt contractSizeLimit) {
+    int frontierContractSizeLimit = contractSizeLimit.orElse(FRONTIER_CONTRACT_SIZE_LIMIT);
```

Comment:
This probably should just be called `contractSizeLimit` since it's not necessarily the frontier limit anymore (similarly for the other cases).